### PR TITLE
Admin login

### PR DESCRIPTION
**Pull request by Nyutonyka**

### CODE REVIEW

**Comment by MariaTarasiuk** [comment]
Pls add description

File: src/main/java/com/store/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @version 0.0.1
+ */
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/admin")
+public class AdminController {
+
+    @PostMapping("/login")
+    public ResponseEntity<String> addProduct(@RequestBody AdminDto adminDto) {
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/AdminController.java:19]: Change method name

File: src/main/java/com/store/controller/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        return new ResponseEntity<>("Hallow Admin!", HttpStatus.CREATED);
+    }
+
+    @GetMapping("/logout")
+    public ResponseEntity<String> getAllProducts() {
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/AdminController.java:24]: Change method name

File: src/main/java/com/store/controller/AdminController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/admin")
+public class AdminController {
+
+    @PostMapping("/login")
+    public ResponseEntity<String> addProduct(@RequestBody AdminDto adminDto) {
+        return new ResponseEntity<>("Hallow Admin!", HttpStatus.CREATED);
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/AdminController.java:20]: Can you add implementation?

File: src/main/java/com/store/controller/AdminController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @GetMapping("/logout")
+    public ResponseEntity<String> getAllProducts() {
+        return new ResponseEntity<>("Good Bay!", HttpStatus.OK);
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/AdminController.java:25]: Can you add implementation?

File: src/main/java/com/store/dto/AdminDto.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import com.store.entity.Role;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+@Data
+@Accessors(chain = true)
+@AllArgsConstructor
+public class AdminDto {
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/dto/AdminDto.java:12]: Where do you plan to use this DTO? If only in login endpoint then call it LoginRequest.

File: src/main/java/com/store/dto/AdminDto.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    private final String userName = "admin";
+    private final String password = "admin";
+    private final Role role = Role.ADMIN;
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/dto/AdminDto.java:16]: If you use it only in login endpoint `role` shouldn't be here.

File: src/main/java/com/store/dto/AdminDto.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+@Data
+@Accessors(chain = true)
+@AllArgsConstructor
+public class AdminDto {
+
+    private final String userName = "admin";
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/dto/AdminDto.java:14]: why do you have default data here? If `userName` or `password` wasn't sent, you need to return bad request (400)

File: src/main/java/com/store/entity/Role.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package com.store.entity;
+
+public enum Role {
+    ADMIN, GUEST, USER
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/entity/Role.java:4]: who is `guest`? How are you going to use it?

File: src/main/java/com/store/dto/AdminDto.java (hunk 7 above)
Comment (Nyutonyka) [reply src/main/java/com/store/dto/AdminDto.java:14]: Redoing all registration

File: src/main/java/com/store/entity/Role.java (hunk 8 above)
Comment (Nyutonyka) [reply src/main/java/com/store/entity/Role.java:4]: Guest role to use when user is not logged in

**Comment by Nyutonyka** [comment]
Added User entity class, UserDto, AdminDto. Creating a database for users. Partially wrote the implementation of requests. With the authorization check, while I understand, I will add the implementation.

**Pull request closed without merging**